core/os/flock/flock.go

```diff
@@ -18,112 +18,134 @@ import (
 	"errors"
 	"os"
 	"path/filepath"
-	"strings"
 	"sync"
 	"time"
 )
 
 var errAlreadyLocked = errors.New("FLock already locked by others")
+var errCannotMakeDir = errors.New("Cannot make directory for FLock files")
 
 const (
-	deviceLockFileSuffix = ".gapid-device-flock"
+	fileSuffix = ".gapid-flock"
+	subDir     = "gapid-flocks"
 )
 
-// RemoveAllDeviceFLockUnderlyingFiles releases all the FLocks by removing all
-// the underlying files, files whose name ends with `deviceLockFileSuffix`.
-func RemoveAllDeviceFLockUnderlyingFiles() error {
-	return filepath.Walk(os.TempDir(), func(path string, info os.FileInfo, err error) error {
-		if !strings.HasSuffix(path, deviceLockFileSuffix) {
-			return nil
-		}
-		if info.IsDir() {
-			return nil
-		}
-		return os.Remove(path)
-	})
+// ReleaseAllLocks releases all the FLocks by removing all the underlying files.
+func ReleaseAllLocks() error {
+	dir := filepath.Join(os.TempDir(), subDir)
+	info, err := os.Stat(dir)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	if info.IsDir() {
+		return os.Remove(dir)
+	}
+	return nil
 }
 
-// LockDevice acquires the lock for the device specified by the given serial or
-// name of the device in a Blocking way, returns the DeviceFLock which holds the
-// lock and error.
-func LockDevice(sn string) (*DeviceFLock, error) {
-	l, err := NewDeviceFLock(sn)
+// Lock takes the given string as the file name for the underlying file and
+// acquires the lock backed by that underlying file in a blocking way. Returns
+// the Mutex which holds the lock, and error if there are any.
+func Lock(n string) (*Mutex, error) {
+	m, err := New(n)
 	if err != nil {
 		return nil, err
 	}
-	l.Lock()
-	return l, nil
+	m.Lock()
+	return m, nil
 }
 
-// TryLockDevice creates a DeviceFLock and use it to acquires the lock for the
-// device specified by the given serial or name of the device in a Non-Blocking
-// way, returns the created DeviceFLock, boolean to show whether the lock has
-// been acquired or not, and error.
-func TryLockDevice(sn string) (*DeviceFLock, bool, error) {
-	l, err := NewDeviceFLock(sn)
+// TryLock creates a Mutex with the given string used as the name of the underlying
+// file, and use the Mutex to acqure the lock backed by that underlying file in
+// a non-blocking way. Returns the created Mutex and error if there are any.
+func TryLock(n string) (*Mutex, error) {
+	m, err := New(n)
 	if err != nil {
-		return nil, false, err
+		return nil, err
 	}
-	locked := l.TryLock()
-	return l, locked, nil
+	m.TryLock()
```

Comment:
Is there any possibility that this could fail to lock?
How does it differ from `Lock`?

Reply:
`TryLock()` is not blocking, `Lock()` is a blocking call. They both returns the created Mutex